.github/workflows/_tests-on-pr.yml

```diff
@@ -64,6 +68,14 @@ jobs:
         run: |
           if ${{ inputs.headless }}; then
             export DISPLAY=:99
-
           fi
-          python -m pytest
+          pytest --cov
+          coverage report -m
+          codecov
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
```

Comment:
Uploading to the branch within `Codecov.io` other than `main` to generate the PR comment. This approach ensures that the user-facing main branch coverage remains unaffected during a PR:

I have tested using a dummy package:

`branch`:

https://app.codecov.io/gh/bobleesj-test-org/bobleesj.release/commits/test-codecov

`main`:

https://app.codecov.io/gh/bobleesj-test-org/bobleesj.release/commits/main